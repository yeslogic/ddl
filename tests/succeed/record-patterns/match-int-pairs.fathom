let f : U32 -> U32 -> U32
= fun x y =>
let p = (x, y);
match p {
    (0, a) => a + (1 : U32),
    (b, 0) => b + (2 : U32),
    (1, c) => c + (3 : U32),
    (d, 1) => d + (4 : U32),
    (_, _) =>     (5 : U32),
};
{}
