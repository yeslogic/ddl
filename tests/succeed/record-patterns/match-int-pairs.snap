stdout = '''
let f : U32 -> U32 -> U32 = fun x y => let p : (U32, U32) = (x, y);
match p._0 {
    0 => let a : U32 = p._1;
    a + (1 : U32),
    1 => match p._1 {
        0 => let b : U32 = p._0;
        b + (2 : U32),
        1 => let c : U32 = p._1;
        c + (3 : U32),
        _ => let c : U32 = p._1;
        c + (3 : U32),
    },
    _ => match p._1 {
        0 => let b : U32 = p._0;
        b + (2 : U32),
        1 => let d : U32 = p._0;
        d + (4 : U32),
        _ => 5,
    },
};
() : ()
'''
stderr = ''
